### PR TITLE
Add ``cleanup_data`` command

### PR DESCRIPTION
**Pull request by jwhitlock**
Add a command `./manage.py cleanup_data` that detects and optionally cleans data issues. This version has three cleaners:

* ``many-profiles`` - Counts users with multiple profiles (can't fix the problem)
* ``missing-profile`` - Ensures users have a profile
* ``server-storage`` - Ensure no data is stored when ``server_storage=False``

It has passing tests that cover the new lines of code, and passes `mypy` and `black`

Things to try:

* `./manage.py cleanup_data` - Report local issues
* `./manage.py cleanup_data --verbosity=2` - Detailed report
* `./manage.py cleanup_data --clean --verbosity=2` - Detailed report and fix issues

This could also be tried on the development server:

```sh
heroku login
git push heroku HEAD:master  # Deploy
heroku run "./manage.py cleanup_data"
```

The command runs a set of tasks derived from `privaterelay.cleaners.DataIssueTask`. These tasks implement three methods:

* `_get_counts_and_data()` - Query the database for data issues and context, and return a dictionary of counts (`Counts`) and a dictionary of querysets needed for optional cleaning (`CleanupData`)
* `_clean()` - Use the `CleanupData`, stored on the instance, to fix the detected issues
* `markdown_report()` - Create a markdown-formatted subreport of the actions taken, based on `Counts`.

Two base classes, `CleanerTask` and `DetectorTask`, are used for tasks that can and cannot clean the issues they detect.

The command runs each task (or single tasks if requested), summarize the issues detected and cleaned, and output a markdown report of what was done. Additional checks can be added by implementing a `DataIssueTask` and appending it to `privaterelay.management.commands.cleanup_data.Command.task_list`.

I've also moved some user-management commands to the new `privaterelay.management.commands`, to distinguish them from those involving the `emails` app. The profile is in `emails`, so there is a little mixing of areas of concern, but maybe we can change that in the future.

This was originally filed as PR #1994.

### CODE REVIEW

**Review by groovecoder — APPROVED** [review]
Code looks good - just a few nits and questions.

And tests pass and increases test coverage to 78% !!

File: emails/tests/cleaners_tests.py [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+from emails.models import DomainAddress, Profile, RelayAddress
+
+from .models_tests import make_premium_test_user, make_storageless_test_user
+
+
+def setup_server_storage_test_data(add_user_no_storage=False, add_address_data=False):
+    """Setup users and addresses for testing."""
+
+    # Create a user with server storage and addresses
+    user_storage = make_premium_test_user()
```
[the remark this answers is not in the document]
Comment (jwhitlock) [reply emails/tests/cleaners_tests.py]: > nit: `user_storage` sounds a bit off for this. but then `user_with_server_storage` is a bit long, so not sure if there's any better variable name anyway.

`user_with_server_storage` and `user_without_storage` worked well and didn't expand too many lines.

File: emails/tests/cleaners_tests.py [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+Relay Addresses:
+  All: 0
+Domain Addresses:
+  All: 0"""
+
+
+@pytest.mark.django_db
+def test_server_storage_cleaner_some_server_storage() -> None:
+    """ServerStorageCleaner detects that when some users have server storage."""
+    """The command detects when some users have server storage."""
```
[the remark this answers is not in the document]
Comment (jwhitlock) [reply emails/tests/cleaners_tests.py]: > question/suggestion: These comments seem redundant? Do we need them both?

Nope, got distracted while re-writing docstrings from "command" to "ServerStorageCleaner". Good catch!

File: privaterelay/tests/mgmt_cleanup_data_tests.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+from emails.tests.cleaners_tests import setup_profile_mismatch_test_data
+
+COMMAND_NAME = "cleanup_data"
+MOCK_BASE = f"private_relay.management.commands.{COMMAND_NAME}"
+CLEANERS = {"server-storage", "missing-profile", "many-profiles"}
+KNOWN_CLEANER = "server-storage"
+
+
+@pytest.mark.django_db
+def test_dry_run(caplog) -> None:
```
[the remark this answers is not in the document]
Comment (jwhitlock) [reply privaterelay/tests/mgmt_cleanup_data_tests.py:22]: It's one of the built-in fixtures - run `pytest --fixtures` for the full list, or see the [fixtures reference](https://docs.pytest.org/en/7.1.x/reference/fixtures.html) which has some diagrams about fixture discovery in `conftest.py` and plugins

**Review by jwhitlock — COMMENTED** [review]
Thanks @groovecoder, I made the suggested changes, and the code is a bit clearer.

**Pull request merged**